src/anemoi/training/train/forecaster.py

```diff
@@ -219,9 +241,26 @@ def _step(
         validation_mode: bool = False,
     ) -> tuple[torch.Tensor, Mapping[str, torch.Tensor]]:
         del batch_idx
+
+        # preprocess batch and broadcast from reader_group rank 0 to reader_group
+        if self.reader_group_rank == 0:
+            # for validation not normalized in-place because remappers cannot be applied in-place
+            batch = self.model.pre_processors(batch, in_place=not validation_mode)
+        else:
+            # init batch tensor with correct shape on non-root ranks
+            shape = (batch.shape[0],) + tuple(batch[0].tolist())
+            batch = torch.zeros(shape, device=self.device)
```

Comment:
why not torch.empty? Does probably not matter ...

Reply:
You're right, no reason to use zeros instead of empty, changed it.